contracts/periphery/contracts/s2s/calls/PangolinCalls.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0;
+
+import "./share/SystemCalls.sol";
+import "../types/PalletEthereum.sol";
+import "@darwinia/contracts-utils/contracts/SafeMath.sol";
+
+library PangolinCalls {
+    using SafeMath for uint256;
+
+    // According to the EVM gas benchmark, 1 gas ~= 40_000 weight.
+    uint64 public constant WEIGHT_PER_GAS = 40_000;
+
+    // The cap limitation for gas limit
+    // https://github.com/darwinia-network/darwinia-messages-substrate/issues/107#issuecomment-1164185135
+    uint64 public constant MAX_GAS_LIMIT  = 100_000_000_000;
```

Comment:
Gas limit seems to too large？

Reply:
I get it from rust code. not right?

https://github.com/darwinia-network/darwinia-common/blob/eaf8704a2b40d4facb800cc98852772b144de346/node/runtime/pangolin/src/pallets/bridge_dispatch.rs#L26

Reply:
GAS !== Balance

Reply:
```
gas_limit * min_gas_price should <= maxUsableBalance * 10**9
gas_limit                 should <= maxUsableBalance * 10**9 / min_gas_price
gas_limit                 should <= maxUsableBalance * 10**9 / 1 GWEI
gas_limit                 should <= 100 * COIN * 10**9 / 1 GWEI
gas_limit                 should <= 100 * 10**9 * 10**9 / 1 GWEI

so,
max_gas_limit = 100 * 10**9
```

Reply:
I am confused about this, Please help to review @HackFisher @xiaoch05

Reply:
fee = gaslimit * gasprice = 100 COIN
here the gaslimit is too large, usually about several million, can't exceed 2000w in ethereum network.

Reply:
COIN = 10**9
Is there a precision here?